# HG changeset patch
# User Rik <rik@octave.org>
# Date 1439168881 25200
#      Sun Aug 09 18:08:01 2015 -0700
# Node ID a76e20f2f156ebd4ddfc60416b78e835ab289f54
# Parent  a63f9cb609a1d7422999b0836c33f0ab00b98e80
Fix IEEE big-endian/little endian conversions in oct-stream (bug #45687).

* oct-stream.cc (convert_and_copy): Call oct_mach_info::native_float_format ()
rather than oct_mach_info::float_format which is an enum rather than a function.

diff --git a/libinterp/corefcn/oct-stream.cc b/libinterp/corefcn/oct-stream.cc
--- a/libinterp/corefcn/oct-stream.cc
+++ b/libinterp/corefcn/oct-stream.cc
@@ -3184,17 +3184,17 @@ convert_and_copy (std::list<void *>& inp
           for (octave_idx_type i = 0; i < input_buf_elts && j < elts_read;
                i++, j++)
             {
               if (swap)
                 swap_bytes<sizeof (SRC_T)> (&data[i]);
               else if (do_float_fmt_conv)
                 do_float_format_conversion (&data[i], sizeof (SRC_T),
                                             1, from_flt_fmt,
-                                            oct_mach_info::float_format ());
+                                            oct_mach_info::native_float_format ());
 
               dst_elt_type tmp (data[i]);
 
               if (do_NA_conv && __lo_ieee_is_old_NA (tmp))
                 tmp = __lo_ieee_replace_old_NA (tmp);
 
               conv_data[j] = tmp;
             }
